lib/jazzy/config.rb

```diff
@@ -400,6 +401,12 @@ def print_attr_description(attr)
       end
     end
 
+    class PathnameArray < Array
```

Comment:
@segiddins After sleeping on it, I came up with an alternative: https://github.com/pcantrell/jazzy/commit/cc96f9b7ba40d86b7ded80756b1bed4349a34d12

This eliminates the subclassing, suits your wish for more explicitness, and avoid the type case that I don't like.

Its big disadvantage is that there’s now a bit more statefulness on the `Config` object: whenever you set a path-based property, it’s immediately expanded based on the current value of the `base_path` instance var, which [changes and then changes back](https://github.com/pcantrell/jazzy/blob/cc96f9b7ba40d86b7ded80756b1bed4349a34d12/lib/jazzy/config.rb#L319-L330) while reading from a config file.

I don’t have strong feelings about this approach vs. the original. Let me know what you think.